### PR TITLE
Fix #224

### LINKED ISSUE #224: Improve validators on barman new component form
- Facebook must be an URL
- Date must be a valid ISO date

### PR DESCRIPTION
**Pull request by kistora**
### Description of your PR

Will close #224

### Pre-review TODO

- [x] PR is rebased on top of `origin/master` 
- [ ] PR provide new tests for new behaviors
- [x] PR doesn't introduce commented out code
- [x] PR has been tested locally on the main use cases
- [ ] Documentation have been updated if necessary

### CODE REVIEW

File: client/src/app/barmen/edit/barman-edit.component.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ export class BarmanEditComponent implements OnInit {
             lastName: new FormControl('', [Validators.required]),
             firstName: new FormControl('', [Validators.required]),
             nickname: new FormControl('', [Validators.required]),
-            facebook: new FormControl(''),
+            facebook: new FormControl('',
+                [Validators.pattern(/^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/)]),
```
Comment (Embraser01) [inline client/src/app/barmen/edit/barman-edit.component.ts]: We should even check for a facebook link! (starting with`https://www.facebook.com/`)

File: client/src/app/barmen/new/barman-new.component.ts [hunk 2]
```diff
@@ -40,7 +40,8 @@ export class BarmanNewComponent implements OnInit {
             lastName: new FormControl('', [Validators.required]),
             firstName: new FormControl('', [Validators.required]),
             nickname: new FormControl('', [Validators.required]),
-            facebook: new FormControl(''),
+            facebook: new FormControl('',
+                [Validators.pattern(/^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/)]),
```
Comment (Embraser01) [inline client/src/app/barmen/new/barman-new.component.ts]: Same here

Comment (kistora) [reply client/src/app/barmen/new/barman-new.component.ts]: Because I was lazy and I found this regex :) Be free to re work on it !

Comment (Embraser01) [reply client/src/app/barmen/new/barman-new.component.ts]: Ok I will do that

**Pull request merged**